woocommerce/woocommerce-ios
# [Privacy Choices] Banner Presentation - Part 1

Part of #9613 

# Why

This PR kicks off the work for handling the privacy choices banner presentation.
Specifically, it shows the banner when:

- The device has EU locale: This will be changed later to the IP country code.
- The user has not saved privacy choices before: There is no way to save privacy choices, this will be done in a separate PR

# How

- Adds a `hasSavedPrivacyBannerSettings` user default setting to save the privacy banner state
- Adds a `PrivacyBannerPresentationUseCase` type that will be responsible for figuring out when the banner should be presented.
- Adds a `PrivacyBannerPresenter` to handle the actual banner presentation.
- Updates `EUCustomsScenarioValidator` to be able to use the EU country codes set there. In the future we may want to extract those country codes into a more general enum/type.

# Demo

https://github.com/woocommerce/woocommerce-ios/assets/562080/00ed3e38-9280-4674-a979-2d18f1441818

# Testing Steps

- Set your device locale to a country not in the EU zone EG: USA
- Launch the app
- See that no banner is presented

----

- Set your device locale to a country  in the EU zone EG: GB
- Launch the app
- See that the privacy banner is presented.

Note: The banner is presented with a little bit of extra space at the bottom. I'll be looking at improving that in a later PR!

---
